SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/RemotePossessionComponent.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/Components/RemotePossessionComponent.h"
+
+#include "Engine/World.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "LoadBalancing/OwnershipLockingPolicy.h"
+
+DEFINE_LOG_CATEGORY(LogRemotePossessionComponent);
+
+URemotePossessionComponent::URemotePossessionComponent(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+	, bPendingDestroy(false)
+{
+	PrimaryComponentTick.bCanEverTick = true;
+	PrimaryComponentTick.bStartWithTickEnabled = true;
+
+#if ENGINE_MINOR_VERSION <= 23
+	bReplicates = true;
+#else
+	SetIsReplicatedByDefault(true);
+#endif
+}
+
+void URemotePossessionComponent::BeginPlay()
+{
+	Super::BeginPlay();
+	if (GetOwner()->HasAuthority())
+	{
+		if (Target == nullptr)
+		{
+			OnInvalidTarget();
+			return;
+		}
+		USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(GetOwner()->GetNetDriver());
```

Comment:
I do not think this to be necessary, to destroy the component if it is locked (or event to check the lock here).
First, in its current state it would prevent local possession. Why destroy the component if the target is locally authoritative and could be possessed?
Second, the actor could be locked because of a quick and transient reason, like an ability being activated for a few seconds. So I would keep this check where it belongs, that is where it was in the load balancing handler, and not destroy the component if locked.